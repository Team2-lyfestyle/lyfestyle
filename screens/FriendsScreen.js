import React from 'react'
import { View, Text, StyleSheet, FlatList, TouchableOpacity } from 'react-native'
import { Tab, Tabs } from 'native-base';
import { TextInput } from 'react-native-gesture-handler';
import { Ionicons } from '@expo/vector-icons';
import dbCaller from '../util/firebase_queries';

const FriendsScreen = (props) => {
  let [friendsList, setFriendsList] = React.useState([]);
  let [friendRequestsRcvd, setFriendRequestsRcvd] = React.useState([]);
  let [friendRequestsSent, setFriendRequestsSent] = React.useState([]);

  const loadData = async () => {
    let friendsList = await props.route.params.friendService.getFriendsListAsArray();
    let friendRequestsRcvd = await props.route.params.friendService.getFriendRequestsRcvd();
    let friendRequestsSent = await props.route.params.friendService.getFriendRequestsSent();
    setFriendsList(friendsList);
    setFriendRequestsRcvd(friendRequestsRcvd);
    setFriendRequestsSent(friendRequestsSent);
  };

  React.useEffect(() => {
    let callback = () => { loadData(); };
    dbCaller.listenForUserChanges(callback);
    loadData();
    return () => {
      dbCaller.stopListenForUserChanges(callback);
    }
  }, []);

  const renderItem = ({ item, buttons }) => {
    //console.log('render item:', item);
    return (
      <View style={{backgroundColor: '#EFEdF4'}}>
        <View style={{flex: 1, margin: 15, height: 80, borderRadius: 8,
              backgroundColor: 'white', flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center'}}>
          <Text style={{fontSize: 20, marginLeft: 20}}>
            {item.name}
          </Text>
          {buttons.map( button => (
            <TouchableOpacity
              style={{backgroundColor: '#406081', marginRight: 20, borderRadius: 10}}
              key={button.buttonName}
              onPress={() => {button.onPress(item.uid)}}
            >
              <Text style={{color: '#00FED4', margin: 15, fontSize: 18}}>{button.buttonName}</Text>
            </TouchableOpacity>
          ))}
        </View>
      </View>
    )
  };

  const friendsButtons = [{buttonName: 'Remove', onPress: (uid) => {props.route.params.friendService.removeFriend(uid); loadData();}}];
  const sentButtons = [{buttonName: 'Delete', onPress: (uid) => {props.route.params.friendService.deleteSentFriendRequest(uid); loadData();}}];
  const receivedButtons = [{buttonName: 'Accept', onPress: (uid) => {props.route.params.friendService.acceptFriendRequest(uid); loadData();}},
                           {buttonName: 'Deny', onPress: (uid) => {props.route.params.friendService.deleteRcvdFriendRequest(uid); loadData();}}];
  return (
    <View style={styles.container}>
      <Tabs>
        <Tab heading="Friends">
          <FlatList
            style={styles.flatList}
            data={friendsList}
            renderItem={({item}) => renderItem({item: item, buttons: friendsButtons}) }
            keyExtractor={(item) => item.uid}
          />
        </Tab>
        <Tab heading="Sent">
          <FlatList
            style={styles.flatList}
            data={friendRequestsSent}
            renderItem={({item}) => renderItem({item: item, buttons: sentButtons}) }
            keyExtractor={(item) => item.uid}
          />
        </Tab>
        <Tab heading="Received">
          <FlatList
            style={styles.flatList}
            data={friendRequestsRcvd}
            renderItem={({item}) => renderItem({item: item, buttons: receivedButtons}) }
            keyExtractor={(item) => item.uid}
          />
        </Tab>
      </Tabs>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#EFEdF4',
  },
  header: {
    paddingTop: 40,
    paddingBottom: 16,
    backgroundColor: '#204051',
    alignItems: 'center',
    justifyContent: 'center',
    borderBottomWidth: 1,
    borderBottomColor: '#EbECf4',
    shadowColor: '#454D65',
    shadowOffset: { height: 5 },
    shadowRadius: 15,
    shadowOpacity: 0.2,
    zIndex: 10,
  },
  headerTitle: {
    fontSize: 20,
    fontWeight: '500',
    color: 'white'
  },
  flatList: {
    flex: 1,
    backgroundColor: '#EFEdF4'
  }
});

export default FriendsScreen;
