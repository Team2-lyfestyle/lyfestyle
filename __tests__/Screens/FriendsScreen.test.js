import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FriendsScreen from '../../screens/FriendsScreen';

import renderer from 'react-test-renderer';

describe('<FriendsScreen />', () => {
  it('Should render the <FriendsScreen /> component', () => {
    const wrapper = shallow(<FriendsScreen />);
    expect(wrapper.find(FriendsScreen)).toMatchSnapshot();
  });
});

test('renders correctly', () => {
  const tree = renderer.create(<FriendsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
