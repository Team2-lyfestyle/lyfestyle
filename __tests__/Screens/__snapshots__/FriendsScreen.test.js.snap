// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FriendsScreen /> Should render the <FriendsScreen /> component 1`] = `null`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EFEdF4",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 0.5,
            "borderColor": "#a7a6ab",
            "elevation": 3,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-around",
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
          },
          Object {
            "backgroundColor": "#F8F8F8",
          },
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          active={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F8F8F8",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "marginLeft": 7,
                  "marginRight": 7,
                },
                Object {
                  "fontSize": 15,
                },
                undefined,
                Object {
                  "color": "#007aff",
                },
              ]
            }
            uppercase={false}
          >
            Friends
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          active={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F8F8F8",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#6b6b6b",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": 7,
                  "marginRight": 7,
                },
                Object {
                  "fontSize": 15,
                },
                undefined,
                Object {
                  "color": "#6b6b6b",
                },
              ]
            }
            uppercase={false}
          >
            Sent
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          active={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F8F8F8",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#6b6b6b",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": 7,
                  "marginRight": 7,
                },
                Object {
                  "fontSize": 15,
                },
                undefined,
                Object {
                  "color": "#6b6b6b",
                },
              ]
            }
            uppercase={false}
          >
            Received
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#007aff",
            "bottom": 0,
            "height": 4,
            "left": 0,
            "position": "absolute",
            "width": 250,
          }
        }
      />
    </View>
    <RCTScrollView
      alwaysBounceVertical={false}
      automaticallyAdjustContentInsets={false}
      contentOffset={
        Object {
          "x": 0,
        }
      }
      directionalLockEnabled={true}
      horizontal={true}
      keyboardDismissMode="on-drag"
      keyboardShouldPersistTaps="handled"
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      pagingEnabled={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <View
            heading="Friends"
            style={
              Object {
                "backgroundColor": "#FFF",
                "flex": 1,
              }
            }
          >
            <RCTScrollView
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              style={
                Object {
                  "backgroundColor": "#EFEdF4",
                  "flex": 1,
                }
              }
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View />
            </RCTScrollView>
          </View>
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <View
            heading="Sent"
          />
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        >
          <View
            heading="Received"
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
